### PR TITLE
MO cannot convert Caffe model from PyPI "openvino-dev" distribution

### PR DESCRIPTION
**Pull request by slyubimt**
### Details:
 - *Fix path construction to default Caffe* parser location for wheel directory layout*

### Tickets:
 - *53115*

### CODE REVIEW

File: model-optimizer/mo/utils/cli_parser.py [hunk 1]
```diff
@@ -428,12 +428,12 @@ def get_caffe_cli_parser(parser: argparse.ArgumentParser = None):
     caffe_group.add_argument('--caffe_parser_path',
                              help='Path to Python Caffe* parser generated from caffe.proto',
                              type=str,
-                             default=os.path.join(os.path.dirname(sys.argv[0]), 'mo', 'front', 'caffe', 'proto'),
+                             default=os.path.join(os.path.dirname(__file__), '..', 'mo', 'front', 'caffe', 'proto'),
```
Comment (lazarevevgeny) [inline model-optimizer/mo/utils/cli_parser.py:431]: use `os.pardir` instead of `..`.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

**Comment by lazarevevgeny** [comment]
@slyubimt , have you verified that MO works from both pip and after clone of github?

**Comment by slyubimt** [comment]
@lazarevevgeny, tested including  --caffe_parser_path option

File: model-optimizer/mo/utils/cli_parser.py [hunk 2]
```diff
@@ -428,12 +428,12 @@ def get_caffe_cli_parser(parser: argparse.ArgumentParser = None):
...
                              help='Path to Python Caffe* parser generated from caffe.proto',
                              type=str,
-                             default=os.path.join(os.path.dirname(sys.argv[0]), 'mo', 'front', 'caffe', 'proto'),
+                             default=os.path.join(os.path.dirname(__file__), os.pardir, 'front', 'caffe', 'proto'),
                              action=CanonicalizePathCheckExistenceAction)
     caffe_group.add_argument('-k',
                              help='Path to CustomLayersMapping.xml to register custom layers',
                              type=str,
-                             default=os.path.join(os.path.dirname(sys.argv[0]), 'extensions', 'front', 'caffe',
+                             default=os.path.join(os.path.dirname(__file__), os.pardir, os.pardir, 'extensions', 'front', 'caffe',
```
Comment (dkurt) [inline model-optimizer/mo/utils/cli_parser.py:436]: Is that ok that the same sys.argv[0] changed to __file__+pardir in one parameter but two pardir in another?

Comment (slyubimt) [reply model-optimizer/mo/utils/cli_parser.py:436]: cli_parser.py resides on different levels for dirs:

.\mo\utils\cli_parser.py: 

- .\mo\front\caffe\proto
- .\extensions\front\caffe

**Comment by lazarevevgeny** [comment]
@dkurt , @slyubimt  I recommend to add one Caffe model with a custom proto parser to the pre-commit for pypi distribution. This could be done in a separate PR.

**Review by dkurt — APPROVED** [review]

**Review by sadolini — APPROVED** [review]

**Pull request merged**